cli/command/image/push.go

```diff
@@ -48,12 +52,32 @@ func NewPushCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVarP(&opts.all, "all-tags", "a", false, "Push all tags of an image to the repository")
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Suppress verbose output")
 	command.AddTrustSigningFlags(flags, &opts.untrusted, dockerCli.ContentTrustEnabled())
+	flags.StringVar(&opts.platform, "platform", os.Getenv("DOCKER_DEFAULT_PLATFORM"),
+		`Push a platform-specific manifest as a single-platform image to the registry.
+'os[/arch[/variant]]': Explicit platform (eg. linux/amd64)`)
+	flags.SetAnnotation("platform", "version", []string{"1.46"})
 
 	return cmd
 }
 
 // RunPush performs a push against the engine based on the specified options
 func RunPush(ctx context.Context, dockerCli command.Cli, opts pushOptions) error {
+	var platform *platforms.Platform
+	if opts.platform != "" {
```

Comment:
Not for this PR, and not urgent, but if we decide to add more locations to pass `--platform`, and we want to consume those as a `platforms, we could consider implementing a platform _option_ that can be used for flags, performs the validation as part of that, and directly setting a `Platform`.

Reply:
Yeah I think we should, but left that to a second PR that will reuse this logic.

Btw, we already have it, but for the "string" platform: https://github.com/docker/cli/blob/0022fe711170b7e15daf5f75bd14074e423f5f5a/cli/command/utils.go#L158 🙈

Reply:
Ah! We should look where/how that's used. If the result is only "internal", it could still make sense to change it, and have the consumer convert it to a string (where needed 🤔).

In either case; all for separate work, just that I thought of it.